htlcswitch/link.go

```diff
@@ -554,8 +565,14 @@ func (l *channelLink) sampleNetworkFee() (lnwallet.SatPerKWeight, error) {
 // shouldAdjustCommitFee returns true if we should update our commitment fee to
 // match that of the network fee. We'll only update our commitment fee if the
 // network fee is +/- 10% to our network fee.
-func shouldAdjustCommitFee(netFee, chanFee lnwallet.SatPerKWeight) bool {
+func shouldAdjustCommitFee(netFee, chanFee lnwallet.SatPerKWeight,
+	maxFee btcutil.Amount) bool {
+
 	switch {
+	// The network fee should not exceed our maximum fee allowed.
+	case lnwallet.CalcFee(netFee) > maxFee:
+		return false
```

Comment:
`maxFee` depends on our balance and is recomputed before every `shouldAdjustCommitFee` call, so this should already be handled if our balance decreases?

Reply:
I was thinking of this scenario:

* `netFee` is constant at 1000
* `maxFeeAllocation` is 50%
* `AvailableBalance()` is 10000 sats
* current `commitFee` is 1000

All is good, as we are below 50%.

* Then we spend 9500 sats, `AvailableBalance()` is 500 sats
* `balance = AvailableBalance() + current_commit_fee` = 1500 sats
* `maxFee = 1500 * 50% = 750 sats`

I'd say the commit fee should now be updated to 750 sats, but in the code above you return early because `1000 > 750`?

Reply:
Ah, good point. The scope of the issue to me seemed just to avoid proposing the fee updates instead of limiting it to the max. Though I guess limiting it to the max would be beneficial as it puts us closer to the network fee if our current fee is way below it.

Reply:
FWIW, in your example @joostjager, we wouldn't be able to have an available balance of 500 satoshis, as that's below our reserve.

Reply:
With that aside, I think example stands, but it's out of the scope of the initial target here (just to limit how high we'll go).

Reply:
Why is it out of scope? Isn't it better to keep sending updates, but cap them to the max?

Reply:
Yes, that is what I thought indeed, `min(maxFee, commitFee)`.